molgenis-data/src/main/java/org/molgenis/data/support/DefaultAttributeMetaData.java

```diff
@@ -23,8 +47,10 @@
  */
 public class DefaultAttributeMetaData implements AttributeMetaData
 {
-	private String name;
-	private FieldType fieldType = MolgenisFieldTypes.STRING;
+	private Map<String, AttributeChangeListener> changeListeners;
+
+	private final String name;
+	private FieldType fieldType;
 	private String description;
 	private boolean nillable = true;
 	private boolean readOnly = false;
```

Comment:
This is a non-trivial task. I created [this story](http://www.molgenis.org/ticket/3933) to remove these fields.